rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
My first impression is negative duration. I think part of the reason `select()` does what it does is because it doesn't have the ability to store negative time values.